extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ProxyConfig.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.vertx.http.runtime;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+/**
+ * Holds configuration related with proxy addressing forward.
+ */
+@ConfigGroup
+public class ProxyConfig {
```

Comment:
I think the existing proxy config should be moved to this class as well (proxyAddressForwarding and allowForwarded).

To do this mark the original ones deprecated, change them to Optional<Boolean>, and then add new ones here. If the old ones are set then log a warning about deprecation and use those values, otherwise use the new config.

Reply:
done